File: Homeworks/C#2/StringsAndTextProcessingHomework/12.ParseURL/ParseURL.cs
/*
Problem 12. Parse URL

Write a program that parses an URL address given in the format: [protocol]://[server]/[resource] and extracts from it the [protocol], [server] and [resource] elements.
Example:

                       URL                                               Information
http://telerikacademy.com/Courses/Courses/Details/212	              [protocol] = http 
                                                                  [server] = telerikacademy.com 
                                                            [resource] = /Courses/Courses/Details/212 
*/

using System;

namespace ParseURLProblem   
{
    class ParseURL
    {
        static void Main()
        {
            Console.Write("Please enter URL addres: ");
            string url = Console.ReadLine();

            string protocol = url.Substring(0, url.IndexOf("://"));
            url = url.Substring(url.IndexOf("://") + 3);
            string server = url.Substring(0, url.IndexOf("/"));
            string resource = url.Substring(url.IndexOf("/"));
            Console.WriteLine("[protocol] = {0}\n[server] = {1}\n[resource] = {2}", protocol, server, resource);
        }
    }
}

File: Homeworks/JavaScriptFundamentals/08.Strings/07.ParseURL/ParseURL.js
 /*
  Write a script that parses an URL address given in the format: [protocol]://[server]/[resource] and extracts from it the [protocol], [server] and [resource] elements.
  Return the elements in a JSON object.
  Example:

  URL:
  http://telerikacademy.com/Courses/Courses/Details/239

  result:
  protocol: http
  server: telerikacademy.com
  resource: /Courses/Courses/Details/239
 */

 var jsConsole,
     url = 'http://telerikacademy.com/Courses/Courses/Details/239';


 function parseURL(url) {
     var protocol = '',
         server = '',
         resource = '';

     protocol = url.substr(0, url.indexOf("://"));
     url = url.substr(url.indexOf("://") + 3);
     server = url.substr(0, url.indexOf("/"));
     resource = url.substr(url.indexOf("/"));

     console.log('protocol: ' + protocol + "\n" + 'server: ' + server + "\n" + 'resource: ' + resource);
     jsConsole.writeLine('protocol: ' + protocol);
     jsConsole.writeLine('server: ' + server);
     jsConsole.writeLine('resource: ' + resource);

 }


 parseURL(url);